eng/source-build-release.yml

```diff
@@ -0,0 +1,127 @@
+    type: boolean
+    default: false
+  - name: customTag
+    displayName: Installer custom tag
+    type: string
+    default: v6.0.XYY-source-build
+
+variables:
+  repoId: ''
+  discussionCategoryId: ''
+  announcementOrg: 'dotnet'
+  announcementRepo: 'source-build'
+  sourceRepoUrl: 'https://github.com/dotnet/installer.git'
+
+stages:
+
+- stage: Setup
+  jobs:
+  - job: SetupJob
+    displayName: Setup
+    steps:
+    - script: |
+        set -euxo pipefail
+        if [ "${{ parameters.useCustomTag }}" = "True" ] ; then
+          tag=${{ parameters.customTag }}
+          echo "Using custom tag $(tag)"
+        else
+          tag=v${{ parameters.sdkVersion }}
+          echo "Using tag $(tag)"
+        fi
+        echo "##vso[task.setvariable variable=Tag;isOutput=true]${tag}"
+      name: SetTag
+      displayName: Set tag
+    - script: |
+        set -euxo pipefail
+        tag="$(SetTag.Tag)"
+        git clone ${{ variables.sourceRepoUrl }} source
+        cd source
+        git show-ref --tags
+        if git rev-parse "$tag" >/dev/null 2>&1; then
+          echo "Tag $tag exists.";
+        else
+          echo "Tag $tag does not exist."
+          exit 1
+        fi
+      displayName: Ensure tag exists
+
+- stage: CreateReleaseAnnouncement
+  displayName: Create Release Announcement
+  dependsOn: Setup
+  jobs:
+  - job: CreateReleaseAnnouncementJob
+    displayName: Create Release Announcement
+    variables:
+    - name: Tag
+      value: $[ stageDependencies.Setup.SetupJob.outputs['SetTag.Tag'] ]
+    steps:
+
+    - script: |
+        set -euxo pipefail
+        query='query { repository(owner: "${{ variables.announcementOrg }}", name: "${{ variables.announcementRepo }}") { id } }'
+        repo_id=$( gh api graphql -f query="$query" --template '{{.data.repository.id}}' )
+        echo ${{ variables.announcementOrg }}/${{ variables.announcementRepo }} repo ID is ${repo_id}
+        echo "##vso[task.setvariable variable=repoId;]${repo_id}"
+      displayName: Get repo ID
+      env:
+        GH_TOKEN: $(BotAccount-dotnet-sb-bot-pat)
+
+    - script: |
+        set -euxo pipefail
+        query='query { repository(name: "${{ variables.announcementRepo }}", owner: "${{ variables.announcementOrg }}") { discussionCategories(first: 10) { edges { node { id, name } } } } }'
+        category_id=$( gh api graphql -f query="$query" --template '{{range .data.repository.discussionCategories.edges}}{{if eq .node.name "Announcements"}}{{.node.id}}{{end}}{{end}}' )
+        echo Discussion Category ID is ${category_id}
+        echo "##vso[task.setvariable variable=discussionCategoryId;]${category_id}"
+      displayName: Get announcement discussion category ID
+      env:
+        GH_TOKEN: $(BotAccount-dotnet-sb-bot-pat)
+
+    - script: |
+        set -euxo pipefail
+        echo Repo ID is $(repoId)
+        echo Discussion Category ID is $(discussionCategoryId)
+
+        # Read runtime version from installer's release commit
```

Comment:
What do you think about moving this logic to the SetupJob.  It would eliminate the need to clone the repo in this job.  The runtime version feels like something that has a high chance of being needed in other jobs down the road.